### DISCUSSION TITLE
Shorcut keyboard

### DISCUSSION THREAD

**Discussion by mlerjen** [OpenOrienteering/mapper#1997] [Show and tell]
Let me show and tell you about my Open Orienteering Mapper Shorcut Keyboard :-)
It consist of a 23 key programmable keyboard equipped with transparent cover keycaps. 
I now can reach all the essential shortcuts single handed reducing both mouse paths  and hand paths (right hand stays on mouse).
As for now I have the follwing shortcuts on these keys:
esc : F10 : F3 : ctrl&shift&9 : ctrl&shift&8 : del
P : E : L: F : G : ctrl&Z
shift : D : C : U : K : backtab
ctrl : X : V : space : enter

![20211028_130522](https://user-images.githubusercontent.com/15611653/139288266-e23c13de-57f4-4cef-a3dd-41a2f3fc8aaa.jpg)

Can it get better? - Hardly.

Except... 
1) X is a still unused shortcut. I would love to have it on the "cut away"
2) Is it technically possible that the Android Version listens to shortcuts too? - I tested and found that it listens to keys like crtl, shift, space and delete. Would it be possible for keys too? - I could think of a minimal bluetooth keyboard programmed with the most essential keys. ATM I think for me it would be "jump to my actual position" and "open/close template dialog" :-)
![image](https://user-images.githubusercontent.com/15611653/139290507-4e88076f-2171-459b-81cc-924c88f2da15.png)

**Comment by lpechacek** [comment]
> 2\. Is it technically possible that the Android Version listens to shortcuts too?

IMO yes. There is a custom filter in the Android version that tweaks keyboard input (to handle the Android Back key but also possibly blocking other keyboard input). If the filter is tuned to let the shortcut keys pass through, the shortcuts may work.

As a side note, my long term vision is to allow the desktop UI on Android. Some of the contemporary devices can connect keyboard, mouse and video through the USB-C connector (e.g. Samsung Galaxy Tab Active3 has this capability - it's called DEX). That means field work can be done on the internal display and the map part can be finished on the same device connected to a desktop setup (keyboard, mouse, external display) in the evening. No need for a laptop for desktop work and file transfer between laptop and tablet.
